__tests__/app.test.js

```diff
@@ -345,4 +345,27 @@ describe("GET /api/articles", () => {
         expect(body.msg).toBe("Topic does not exist");
       });
   });
+  test("GET: 200 sends an array with an empty topic as it is not attached to any articles", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toEqual([])
+        });
+      });
+  });
+
+
+describe("GET /api/articles/:article_id", () => {
+  test("GET: 200 responds with the selected article with a comment count", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.article.length).toBe(1);
```

Comment:
Body.article should not be an array, but the article object itself. So we would not need a loop. At the moment you are trying to loop through an array with 1 item, which is unnecessary.

Please see my comment in the model and refactor your code, so in your response on the key of 'article' you will have one article object, instead of an array with 1 item.